calliope/backend/pyomo/constraints/energy_balance.py

```diff
@@ -444,6 +450,22 @@ def balance_supply_plus_constraint_rule(backend_model, loc_tech, timestep):
         )
 
 
+def storage_dod_constraint_rule(backend_model, loc_tech, timestep):
+    """
+    Forces storage state of charge to be greater than the allowed depth of discharge.
+
+    .. container:: scrolling-wrapper
+
+        .. math::
+
+            \\boldsymbol{storage}(loc::tech, timestep) >=
+            \\boldsymbol{storage_DoD}\\forall loc::tech \\in loc::techs_{storage}, \\forall timestep \\in timesteps
+
+    """
+    storage_dod = get_param(backend_model, 'storage_dod', loc_tech)
+    return backend_model.storage[loc_tech, timestep] >= storage_dod
```

Comment:
I'm generally unsure as to the use of this constraint. It would force a storage device to be kept constantly topped up to `depth_of_discharge * storage_cap`, leading to odd spikes in the timeseries when some carrier production has to be diverted to keep the storage at this level. I suppose this is the only way of avoiding binary variables?

As a result, I would move this constraint to `dispatch.py`, since that is where the other constraint linking `storage` and `storage_cap` is.